api/database/chains_test.go

```diff
@@ -0,0 +1,389 @@
+package database_test
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/emerishq/demeris-api-server/api/database"
+	utils "github.com/emerishq/demeris-api-server/api/test_utils"
+	"github.com/emerishq/demeris-backend-models/cns"
+	"github.com/stretchr/testify/require"
+)
+
+func (s *TestSuite) TestChain() {
+	tests := []struct {
+		name      string
+		chainName string
+		expRes    cns.Chain
+		success   bool
+	}{
+		{
+			"chain not found",
+			"invalidChain",
+			cns.Chain{},
+			false,
+		},
+		{
+			"disabled chain",
+			utils.DisabledChain.ChainName,
+			cns.Chain{},
+			false,
+		},
+		{
+			"inserted chain",
+			utils.ChainWithPublicEndpoints.ChainName,
+			utils.ChainWithPublicEndpoints,
+			true,
+		},
+	}
+
+	for _, tt := range tests {
+		s.Run(tt.name, func() {
+			res, err := s.ctx.Router.DB.Chain(tt.chainName)
+			if tt.success {
+				s.Require().NoError(err)
+				compareAsBytes(s.T(), tt.expRes, res)
+			} else {
+				s.Require().Error(err)
+			}
+		})
+	}
+}
+
+func compareAsBytes(t *testing.T, expected interface{}, actual interface{}) {
+	actualBytes, err := json.Marshal(actual)
+	require.NoError(t, err)
+	expBytes, err := json.Marshal(expected)
+	require.NoError(t, err)
+	require.Equal(t, expBytes, actualBytes)
```

Comment:
`JSONEq` gives a far more readable output in case of diff : 

```suggestion
	require.JSONEq(t, string(expBytes), string(actualBytes))
```

Example where I intentionally introduced a change on the `Logo` field :
![image](https://user-images.githubusercontent.com/92280/165758040-d3424a59-b47b-432e-98e2-93b72b9ca72a.png)

Compared to the output of `Equal` with bytes:
![image](https://user-images.githubusercontent.com/92280/165760183-624c5d44-d7a2-40a2-8c2c-e91bc3de1c69.png)